File: src/main/java/cz/kucharo2/service/CashDeskService.java
package cz.kucharo2.service;

import cz.kucharo2.data.entity.Account;
import cz.kucharo2.data.entity.Order;
import cz.kucharo2.data.entity.OrderItem;
import cz.kucharo2.data.entity.RestaurantTable;

import java.util.List;

/**
 * Created by Roman on 12/2/2014.
 */
public interface CashDeskService {

	/**
	 * Create order for specific table
	 *
	 * @param table table
	 * @return id of created order
	 */
	Order createOrderOnTable(RestaurantTable table);

	/**
	 * Create order item of food
	 *
	 * @param orderItemFoods order item to be created
	 */
	void createOrderItem(OrderItem orderItemFoods);

	/**
	 * Returns unpaid order items of food for specific order
	 *
	 * @param orderId order id
	 * @return unpaid order items
	 */
	List<OrderItem> getUnpaidOrderItemFoodByOrder(Integer orderId);

	/**
	 * Returns unpaid finished order items for specific table and specific user
	 *
	 * @param tableId table id
	 * @return unpaid finished orders on given table
	 */
	List<OrderItem> getUnpaidFinishedOrderItemsOnTable(Integer tableId, Integer userId);

	/**
	 * Returns users which have unpaid order items on specific table
	 * @param tableId table id
	 * @return users having unpaid order items on table
	 */
	List<Account> getUsersHavingUnpaidFinishedOrdersOnTable(Integer tableId);

	/**
	 * Pay order items
	 * @param orderItemsIds order item ids of dishes to be paid
	 */
	void pay(List<Integer> orderItemsIds);

}

File: src/main/webapp/js/services/cashDeskService.js
/**
 * Angular service for payments
 * @type {angular.service}
 * @author Pavel Matyáš (matyapav@fel.cvut.cz)
 */
app.service('CashDeskService', function ($http) {

    /**
     * Gets unpaid order items for logged user
     * @returns {HttpPromise}
     */
    this.getUnpaidOrderItems = function (tableId) {
        return $http.get(apiPrefix + "/cashDesk/" + tableId + "/unpaidOrderItems");
    };

    /**
     * Gets users which have some unpaid order items on specified table
     * @param tableId
     * @returns {HttpPromise}
     */
    this.getUsersHavingUnpaidFinishedOrdersOnTable = function (tableId) {
        var loggedWaiter = localStorage.getItem("loggedWaiter");
        if(loggedWaiter !== null && loggedWaiter !== ""){
            return $http({
               url: apiPrefix + "/cashDesk/" + tableId + "/unpaidUsers",
               method: "GET",
                headers: {
                   "Authorization" : "Basic "+loggedWaiter
                }
            });
        }
        return null;
    };

    /**
     * Gets unpaid order items on specified table which belong to specified user
     * @param tableId
     * @param userId
     * @returns {HttpPromise}
     */
    this.getUnpaidFinishedOrderItemsOnTableForUser = function (tableId, userId) {
        var loggedWaiter = localStorage.getItem("loggedWaiter");
        if(loggedWaiter !== null && loggedWaiter !== ""){
            return $http({
                url: apiPrefix + "/cashDesk/" + tableId + "/unpaidOrderItems/" + userId,
                method: "GET",
                headers: {
                    "Authorization" : "Basic "+loggedWaiter
                }
            });
        }
        return null;
    };

    /**
     * Pays order items by ids
     * @param orderItemsIds - array of integers representing order item ids
     * @returns {*}
     */
    this.payItems = function (orderItemsIds) {
        var loggedWaiter = localStorage.getItem("loggedWaiter");
        if(loggedWaiter !== null && loggedWaiter !== "") {
            return $http({
                url: apiPrefix + "/cashDesk/pay",
                method: "POST",
                headers: {
                    "Content-Type": "application/json",
                    "Authorization": "Basic " + loggedWaiter
                },
                data: orderItemsIds
            });
        }
        return null;
    }

});